### PR TITLE
added support @kphp-json tag

### PR DESCRIPTION
**Pull request by Danil42Russia**
Added support for `@kphp-json` tags from PR VKCOM/kphp#510

### CODE REVIEW

**Review by i582 — APPROVED** [review]
LGTM

File: src/main/kotlin/com/vk/kphpstorm/kphptags/KphpJsonTag.kt [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        return true
+    }
+
+    private fun checkFloatPrecision(
+        propertyPsi: KphpDocJsonPropertyPsiImpl,
+        property: Property,
+        holder: AnnotationHolder,
+        docTag: PhpDocTag
+    ): Boolean {
+        val isUsedCorrectFloatPrecision = (propertyPsi.intValue() ?: 0) > 0
```
Comment (i582) [inline src/main/kotlin/com/vk/kphpstorm/kphptags/KphpJsonTag.kt:222]: But if `value == 0`?

Comment (i582) [reply src/main/kotlin/com/vk/kphpstorm/kphptags/KphpJsonTag.kt:222]: image

Comment (Danil42Russia) [reply src/main/kotlin/com/vk/kphpstorm/kphptags/KphpJsonTag.kt]: fixed!

File: src/main/kotlin/com/vk/kphpstorm/KphpStormParserDefinition.kt [hunk 2]
```diff
@@ -35,6 +35,9 @@ class KphpStormParserDefinition() : PhpParserDefinition() {
             KphpDocTplParameterDeclPsiImpl.elementType    -> KphpDocTplParameterDeclPsiImpl(node)
             KphpDocElementTypes.kphpDocTagWarnPerformance -> KphpDocTagWarnPerformancePsiImpl(node)
             KphpDocWarnPerformanceItemPsiImpl.elementType -> KphpDocWarnPerformanceItemPsiImpl(node)
+            KphpDocElementTypes.kphpDocTagJson            -> KphpDocTagJsonPsiImpl(node)
+            KphpDocJsonPropertyPsiImpl.elementType        -> KphpDocJsonPropertyPsiImpl(node)
```
Comment (unserialize) [inline src/main/kotlin/com/vk/kphpstorm/KphpStormParserDefinition.kt:39]: Here and everywhere: not "property", but "attrubite" (or just "attr") — to match with PHP/KPHP

File: src/main/kotlin/com/vk/kphpstorm/typeProviders/JsonEncoderTypeProvider.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            return null
+        }
+
+        if (classRef.name != "JsonEncoder") {
+            val phpClass = classRef.resolve()
+            if (phpClass !is PhpClass) {
+                return null
+            }
+
+            val referenceElement = phpClass.extendsList.referenceElements
```
Comment (unserialize) [inline src/main/kotlin/com/vk/kphpstorm/typeProviders/JsonEncoderTypeProvider.kt:43]: I suppose, that multi-level inheritance from JsonEncoder won't be handled here

Comment (i582) [reply src/main/kotlin/com/vk/kphpstorm/typeProviders/JsonEncoderTypeProvider.kt:43]: and it seems the name of the variable does not reflect that there are several elements (list)

File: src/main/kotlin/com/vk/kphpstorm/KphpStormParserDefinition.kt (hunk 2 above)
Comment (Danil42Russia) [reply src/main/kotlin/com/vk/kphpstorm/KphpStormParserDefinition.kt]: done

**Pull request closed without merging**